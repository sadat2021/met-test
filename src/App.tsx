import React from 'react';

function App() {
  return (
    <div className="App">
      met
    </div>
  );
}

export default App;
