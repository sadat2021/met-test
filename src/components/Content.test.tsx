import React from "react";
import { render, screen } from "@testing-library/react";
import App from "./Content";

describe("Content Component", () => {
  test("Show error correctly", () => {
    const { getByTestId, container } = render(
      <App endListLoading={false} error={true} loading={false} items={[]} />
    );
    expect(getByTestId("error")).toBeInTheDocument();
    expect(container).toMatchSnapshot();
  });
  test("Show loading correctly", () => {
    const { getByTestId, container } = render(
      <App endListLoading={false} error={false} loading={true} items={[]} />
    );
    expect(getByTestId("loading")).toBeInTheDocument();
    expect(container).toMatchSnapshot();
  });
  test("Show items correctly", () => {
    const mockItems = [
      {
        status: 200,
        data: {
          title: "title1",
          department: "d1",
          primaryImage: "pi1",
          objectID: 1,
        },
      },
      {
        status: 200,
        data: {
          title: "title2",
          department: "d2",
          primaryImage: "pi2",
          objectID: 2,
        },
      },
      {
        status: 200,
        data: {
          title: "title3",
          department: "d3",
          primaryImage: "pi3",
          objectID: 3,
        },
      },
    ];
    const { getAllByTestId, container } = render(
      <App
        endListLoading={false}
        error={false}
        loading={false}
        items={mockItems}
      />
    );
    expect(getAllByTestId("item").length).toBe(3);
    expect(container).toMatchSnapshot();
  });
});
