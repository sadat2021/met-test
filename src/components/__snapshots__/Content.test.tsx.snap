// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Content Component Show error correctly 1`] = `
<div>
  <div
    class="flex flex-row flex-wrap"
  >
    <div
      class="container p-2 bg-red-500 text-white rounded "
      data-testid="error"
    >
      An error occurred while receiving the Data.
    </div>
  </div>
</div>
`;

exports[`Content Component Show items correctly 1`] = `
<div>
  <div
    class="flex flex-row flex-wrap"
  >
    <div
      class="w-full xl:w-1/4 lg:w-1/3 md:w-1/2 sm:w-1/2 p-4"
      data-testid="item"
    >
      <div
        class="container flex flex-col bg-white shadow-md rounded-md cursor-pointer"
      >
        <img
          alt="title1"
          class="w-full rounded-t-md image"
          src="pi1"
        />
        <h3
          class="m-2 text-lg font-bold"
        >
          title1
          
        </h3>
        <div
          class="flex flex-row flex-wrap mx-2 mb-2"
        >
          <span
            class="w-1/3 text-sm"
          >
            Department:
          </span>
          <span
            class="w-2/3 text-sm font-bold text-right"
          >
            d1
          </span>
        </div>
      </div>
    </div>
    <div
      class="w-full xl:w-1/4 lg:w-1/3 md:w-1/2 sm:w-1/2 p-4"
      data-testid="item"
    >
      <div
        class="container flex flex-col bg-white shadow-md rounded-md cursor-pointer"
      >
        <img
          alt="title2"
          class="w-full rounded-t-md image"
          src="pi2"
        />
        <h3
          class="m-2 text-lg font-bold"
        >
          title2
          
        </h3>
        <div
          class="flex flex-row flex-wrap mx-2 mb-2"
        >
          <span
            class="w-1/3 text-sm"
          >
            Department:
          </span>
          <span
            class="w-2/3 text-sm font-bold text-right"
          >
            d2
          </span>
        </div>
      </div>
    </div>
    <div
      class="w-full xl:w-1/4 lg:w-1/3 md:w-1/2 sm:w-1/2 p-4"
      data-testid="item"
    >
      <div
        class="container flex flex-col bg-white shadow-md rounded-md cursor-pointer"
      >
        <img
          alt="title3"
          class="w-full rounded-t-md image"
          src="pi3"
        />
        <h3
          class="m-2 text-lg font-bold"
        >
          title3
          
        </h3>
        <div
          class="flex flex-row flex-wrap mx-2 mb-2"
        >
          <span
            class="w-1/3 text-sm"
          >
            Department:
          </span>
          <span
            class="w-2/3 text-sm font-bold text-right"
          >
            d3
          </span>
        </div>
      </div>
    </div>
  </div>
</div>
`;

exports[`Content Component Show loading correctly 1`] = `
<div>
  <div
    class="flex flex-row flex-wrap"
  >
    <div
      class="loader mx-auto"
      data-testid="loading"
    />
  </div>
</div>
`;
